app/services/sign_in/idme_public_key_fetcher.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'sign_in/idme/errors'
+
+module SignIn
+  class IdmePublicKeyFetcher
+    def perform
+      uri = URI(Settings.idme.oauth_public_key_url)
+      response = Net::HTTP.get_response(uri)
+      raise StandardError, 'Failed to connect to ID.me public certificates endpoint' unless response.code.to_i == 200
+
+      rs256_keys = JSON.parse(response.body)['keys'].map do |key|
```

Comment:
You could use `filter_map` here - It's the same functionality as using `map` with `compact`